src/frontend/screens/ObservationEdit/index.tsx

```diff
@@ -75,6 +103,12 @@ export const ObservationEdit: NativeNavigationComponent<'ObservationEdit'> = ({
 
     async function createDraftFromExistingObservation(docId: string) {
       const observation = await projectApi.observation.getByDocId(docId);
+      const audioAttachments =
+        observation.attachments.filter(
+          attachment => attachment.type === 'audio',
+        ) || [];
+      setAudioAttachments(audioAttachments);
+
```

Comment:
Thanks for the work you put into. I am a little worried that the logic is a bit challenging to follow. If I understand correctly,

1. you are pulling attachment of type `audio` and setting it to state as `AudioAttachment` on line 106
2. on every render, all audio attachments are processed, with `useAudioProcessing` to return `audioWithUri` on line 83
3. On every change of `audioWithUri` you run a useEffect on line 85
4. The useEffect stringifies the `audioWithUrl`and the state object `processedAudioRecordings` and comparest them. if they are not the same, you are setting the`processedAudioRecordings` to the Audio with URL
5. The you set the 'processedAudioRecording` as a prop to `existingObservationToDraft` on line 124

I think we could simplify it and make it more maintainable for everyone. Would you be open to pairing up on this, or I can provide async feedback if you would prefer?

Reply:
@ErikSin Yes it is confusing. Sorry. I think I have a more clear solution and am working on a little refactor. Still not quite fixed but I will push up another try soon. If that is no good, I think pairing might be more efficient than async if we can find an overlapping time.